src/ring/adapter/jetty9/handlers/async.clj

```diff
@@ -27,10 +27,10 @@
    ^HttpServletRequest request
    ^HttpServletResponse response]
   (try
-    (let [handler (.ringHandler this)
+    (let [[handler options] (.state this)
           {:as options
            :keys [async-timeout]
-           :or {async-timeout 30000}} (meta handler)
+           :or {async-timeout 30000}} options
```

Comment:
I would try to avoid this kind of destructuring in here...If you're trying to provide a default for `async-timeout`, it should happen earlier. It makes sense not only from a performance standpoint, but also from a code-organisation POV.

Reply:
ah apologies - scrap that entire comment. Instead, you may consider `(:async-timeout options 30000)` below, and do away with this destructuring completely